release/prepareRelease.sh.cmake

```diff
@@ -123,45 +128,46 @@ extract_debug_symbols()
 	popd
 }
 
+strip_debug_symbols_darwin()
+{
+	local FILE=$1
+	if [ ! -f "${FILE}" ] ; then
+		return 0
+	fi
+	pushd ${DESTDIR}
+	i686-apple-darwin10-strip -S $FILE
+	popd
+}
+
+
 mecho --blue "## Extracting debug info from files and saving them into dbg"
 
 # strip out debug symbols into external file
 case "$SYSTEM_NAME" in
 	Darwin)
 		echo "extraction of debug symbols for Apple currently not supported" >&2
-		i686-apple-darwin10-strip -S ${DESTDIR}bin/s25client
-		i686-apple-darwin10-strip -S ${DESTDIR}bin/s25edit
-		i686-apple-darwin10-strip -S ${DESTDIR}lib/driver/video/libvideoSDL.dylib
-		i686-apple-darwin10-strip -S ${DESTDIR}lib/driver/audio/libaudioSDL.dylib
-		i686-apple-darwin10-strip -S ${DESTDIR}libexec/RTTR/s25update
-		i686-apple-darwin10-strip -S ${DESTDIR}libexec/RTTR/sound-convert
-		i686-apple-darwin10-strip -S ${DESTDIR}libexec/RTTR/s-c_resample
-	;;
-	Windows)
-		extract_debug_symbols s25client.exe
-		extract_debug_symbols s25edit.exe
-		extract_debug_symbols driver/video/libvideoWinAPI.dll
-		extract_debug_symbols driver/video/libvideoSDL.dll
-		extract_debug_symbols driver/audio/libaudioSDL.dll
-		extract_debug_symbols RTTR/s25update.exe
-		extract_debug_symbols RTTR/sound-convert.exe
-		extract_debug_symbols RTTR/s-c_resample.exe
+		extract_symbols_func="strip_debug_symbols_darwin"
 	;;
-	Linux|FreeBSD)
-		extract_debug_symbols bin/s25client
-		extract_debug_symbols bin/s25edit
-		extract_debug_symbols share/s25rttr/driver/video/libvideoSDL.so
-		extract_debug_symbols share/s25rttr/driver/audio/libaudioSDL.so
-		extract_debug_symbols libexec/RTTR/s25update
-		extract_debug_symbols libexec/RTTR/sound-convert
-		extract_debug_symbols libexec/RTTR/s-c_resample
+	Windows|Linux|FreeBSD)
+		extract_symbols_func="extract_debug_symbols"
+
 	;;
 	*)
 		echo "$SYSTEM_NAME not supported" >&2
 		exit 1
 	;;
 esac
 
+${extract_symbols_func} ${RTTR_BINDIR}/s25client.exe
+${extract_symbols_func} ${RTTR_BINDIR}/s25edit.exe
+${extract_symbols_func} ${RTTR_DRIVERDIR}/video/libvideoWinAPI.dll
```

Comment:
this will fail on linux

Reply:
perhaps we should add a list per architecture which will be processed - we also have `set -e` so each failing command will interrupt the script with error

Reply:
something like

```

symbol_files=${RTTR_BINDIR}/s25client.exe ${RTTR_BINDIR}/s25edit.exe

...

symbol_files=${RTTR_BINDIR}/s25client ${RTTR_BINDIR}/s25edit

...

for file in symbol_files ; do
    ${extract_symbols_func} $file
done

Reply:
I added a check if the file exists. So it doesn't fail

Reply:
yea but if its missing by accident, the build will not fail

Reply:
Alright. Refactored this and the macOS packaging below. What do you think?